libnmstate/nm/device.py

```diff
@@ -347,6 +351,52 @@ def _delete_connection_callback(src_object, result, user_data):
             'dev={}, error=unknown'.format(devname))
 
 
+def delete_device(nmdev):
+    mainloop = nmclient.mainloop()
+    devpath = nmdev.get_path()
+    mainloop.push_action(_safe_delete_device_async, devpath)
+
+
+def _safe_delete_device_async(devpath):
+    mainloop = nmclient.mainloop()
+    client = nmclient.client()
+    nmdev = client.get_device_by_path(devpath)
+    if not nmdev:
+        # No callback is expected, so we should call the next one.
+        mainloop.execute_next_action()
+        return
+
+    user_data = mainloop, nmdev
+    nmdev.delete_async(
+        mainloop.cancellable,
+        _delete_device_callback,
+        user_data,
+    )
+
+
+def _delete_device_callback(src_object, result, user_data):
+    mainloop, nmdev = user_data
+    try:
+        success = src_object.delete_finish(result)
+    except Exception as e:
```

Comment:
To make it more correct, this should check if the failure indicates that the device was already deleted and then ignore this error.

Reply:
Perhaps you are right, but I do not know how to detect it. Also, at this point, I am not sure if it is ok to ignore the error.. it is a good point to consider I guess.
I think we handle these stuff the same in the profile deletion, so it is a wider issue.